sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnApiDoFnRunner.java

```diff
@@ -272,7 +275,7 @@ public void startBundle() {
   @Override
   public void processElement(WindowedValue<InputT> elem) {
     currentElement = elem;
-    try {
+    try (ScopedSideInputContext ignored = setCurrentSideInputContext(processBundleContext)) {
```

Comment:
Scope the side input access from the end of startBundle to the beginning of finishBundle so that we aren't updating a thread local on each process element call.